src/weather.cpp

```diff
@@ -35,16 +35,16 @@ const efftype_id effect_glare( "glare" );
  */
 void weather_effect::glare()
 {
-    if (PLAYER_OUTSIDE && g->is_in_sunlight(g->u.pos()) &&
-        !g->u.worn_with_flag("SUN_GLASSES") && !g->u.has_bionic("bio_sunglasses")) {
-        if(!g->u.has_effect( effect_glare)) {
-            if (g->u.has_trait("CEPH_VISION")) {
+    if( PLAYER_OUTSIDE && g->is_in_sunlight( g->u.pos() ) && !g->u.in_sleep_state() == true &&
```

Comment:
Why not simply `&& !g->u.in_sleep_state()`, there is no need to compare with `true` at all.